arviz/plots/hdiplot.py

```diff
@@ -109,6 +109,21 @@ def plot_hdi(
         >>> ax = az.plot_hdi(x_data, hdi_data=hdi_data[0], color="r", fill_kwargs={"alpha": .2})
         >>> az.plot_hdi(x_data, hdi_data=hdi_data[1], color="k", ax=ax, fill_kwargs={"alpha": .2})
 
+    ``plot_hdi`` can also be used with Inference Data objects. Here we use the posterior predictive
+    to plot the HDI interval.
+
+    .. plot::
+        :context: close-figs
+
+        >>> X = np.random.normal(0,1,100)
+        >>> Y = np.random.normal(2 + X * 0.5, 0.5, (10,100))
+        >>> X = {"x":X}
+        >>> Y = {"y": Y}
+        >>> dataset = az.from_dict(posterior=Y, constant_data = X)
```

Comment:
```suggestion
        >>> idata = az.from_dict(posterior_predictive={"y" : Y}, constant_data = {"x" : X})
```

I'd recommend using this names instead. `y` is generally used for observed data, and its values will generally be stored in the posterior predictive group instead of the posterior, and I would avoid `dataset` as a name because the object it represents is an `InferenceData` as opposed to an `xr.Dataset` which could be confusing, due to this double meaning of dataset.